src/server/session.ts

```diff
@@ -1904,6 +1905,71 @@ export class Session<TMessage = string> implements EventSender {
         });
     }
 
+    private mapCode(args: protocol.MapCodeRequestArgs): protocol.FileCodeEdits[] {
+        const formatOptions = this.getHostFormatOptions();
+        const preferences = this.getHostPreferences();
+        const projects = new Map<Project, MapCodeDocumentMapping[]>();
+        args.mappings.forEach(mapping => {
+            if (!mapping.file) {
+                // TODO(@zkat): We don't support mapping file-less code yet.
+                return;
+            }

```

Comment:
The protocol isn't going to change, because it's externally defined and already being used by various partner teams, and we simply need to do something basic when this happens. CodeMappers isn't meant to be a fully reliable system. "Best effort" is good enough, and thus, I consider this better than doing nothing at all when a file is not present.

Reply:
It might be defined protocol else where but we are writing protocol for tsserver and its not defined. So we should be careful about what we add as we need to maintain it.

Reply:
As above: we should be careful to understand what the plan is with this protocol, and that will include a "global" view, and code that's coming in without a target file, which may be mapped to any file we decide to map to.

Establishing a one-file-only protocol now means we'll need to throw away the public api later and add a second one just because we didn't want to do it this time around.